File: amazon/height-checker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        
        vector<int> newHeights(heights);
        sort(newHeights.begin(), newHeights.end());
        
        int minMoves = 0;
        
        for (int i = 0; i < (int)heights.size(); ++i) {
            
            if (heights.at(i) != newHeights.at(i)) minMoves++;
        }
        
        return minMoves;
    }
};

File: amazon/height-checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        
        # comparing original list with sorted copy
        heights_sorted = sorted(heights)
        
        # count
        students_to_move = 0
        
        for pos in range(len(heights)):
            if heights[pos] != heights_sorted[pos]:
                students_to_move += 1
                
        return students_to_move
